lib/alog.ex

```diff
@@ -3,16 +3,72 @@ defmodule Alog do
     driver: :postgrex,
     migration_lock: "FOR UPDATE"
 
-  @impl true
-  def supports_ddl_transaction? do
-    true
-  end
+  alias Ecto.Adapters.Postgres, as: EAP
 
   @behaviour Ecto.Adapter.Storage
 
   @impl true
-  defdelegate storage_up(opts), to: Ecto.Adapters.Postgres
+  def supports_ddl_transaction?, do: true
 
   @impl true
-  defdelegate storage_down(opts), to: Ecto.Adapters.Postgres
+  defdelegate storage_up(opts), to: EAP
+
+  @impl true
+  defdelegate storage_down(opts), to: EAP
+
+  # overrides insert/6 defined in Ecto.Adapters.SQL
+  def insert(adapter_meta, %{source: "schema_migrations", prefix: prefix}, params, on_conflict, returning, opts) do
```

Comment:
Just noticed that the `insert` function doesn't have the `@impl true`, do we need to add it as the adapter behaviour will need it?